perfkitbenchmarker/beam_benchmark_helper.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper methods for Apache Beam benchmarks.
+
+This file contains methods which are common to all Beam benchmarks and
+executions.
+"""
+
+import os
+
+from perfkitbenchmarker import dpb_service
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+
+
+# TODO: Find a better place for the maven_binary flag.
+flags.DEFINE_string('maven_binary', 'mvn',
+                    'Set to use a different mvn binary than is on the PATH.')
+flags.DEFINE_string('beam_location', None,
+                    'Location of already checked out Beam codebase.')
+flags.DEFINE_string('beam_it_module', None,
+                    'Module containing integration test.')
+flags.DEFINE_string('beam_it_profile', None,
+                    'Profile to activate integration test.')
+flags.DEFINE_string('git_binary', 'git', 'Path to git binary.')
+flags.DEFINE_string('beam_version', None, 'Version of Beam to download. Use'
+                                          ' tag from Github as value.')
+
+FLAGS = flags.FLAGS
+
+SUPPORTED_RUNNERS = [
+    dpb_service.DATAFLOW,
+]
+
+BEAM_REPO_LOCATION = 'https://github.com/apache/beam.git'
+INSTALL_COMMAND = "{0} clean install -DskipTests -Dcheckstyle.skip=true -P{1}"
+
+
+def InitializeBeamRepo(benchmark_spec):
+  if benchmark_spec.dpb_service.SERVICE_TYPE not in SUPPORTED_RUNNERS:
```

Comment:
- Do you think this comment makes sense for the function, (I thought the linter will make an issues in the absence of a comment, or maybe its only when we bring this into google3).
"In the absence of FLAGS.beam_location, initializes the beam source code base by checking out the repository from github. Specific branch selection is supported."

- Should we validate the directory's presence ?